File: DSA/searching_algorithms/linear_search/python/linear_search.py
#!/bin/env python3

"""
    Implementation of Linear Search Algorithm.
"""

def linear_search(arr, target):
    """Performs a linear search for target in arr. Returns index or -1 if not found."""
    for idx, value in enumerate(arr):
        if value == target:
            return idx
    return -1

File: DSA/searching_algorithms/linear_search/javascript/linearSearch.js
#!/bin/env node

/**
 * Implementation of Linear Search Algorithm.
 * Returns the index of the target or -1 if not found.
 */
function linearSearch(arr, target) {
  for (let i = 0; i < arr.length; i++) {
    if (arr[i] === target) {
      return i;
    }
  }
  return -1;
}
